packages/image_picker/image_picker_for_web/lib/image_resizer_utils.dart

```diff
@@ -0,0 +1,73 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math';
+import 'dart:ui';
+
+///a function that checks if an image needs to be resized or not
+bool imageResizeNeeded(double? maxWidth, double? maxHeight, int? imageQuality) {
+  bool resizeNeeded =
+      (maxWidth != null || maxHeight != null || imageQuality != null);
+  if (resizeNeeded) {
+    if (imageQuality != null) {
+      return isImageQualityValid(imageQuality);
+    } else {
+      return true;
+    }
+  } else {
+    return false;
+  }
+}
+
+/// a function that checks if image quality is between 0 to 100
+bool isImageQualityValid(int imageQuality) {
+  return (imageQuality >= 0 && imageQuality <= 100);
+}
+
+/// a functions that calculates the size of the scaled image.
+/// imageWidth is the width of the image
+/// imageHeight is the height of  the image
+/// maxWidth is the maximum width of the scaled image
+/// maxHeight is the maximum height of the scaled image
+Size calculateSizeOfScaledImage(double imageWidth, double imageHeight,
+    double? maxWidth, double? maxHeight) {
```

Comment:
What is the behavior here when both maxWidth and maxHeight are passed? Constrain to fit? Stretch to fit?

Reply:
If we never stretch, but constrain to the smallest side maitaining aspect ratio, the code can be simplified a lot, I think!

Reply:
We are never stretching. My implementation of this function was based on how  the ios side of this plugin scales the image.